docs/specs/future-of-workload.md

```diff
@@ -0,0 +1,43 @@
+# Design proposal for the .NET Aspire workload going forward
```

Comment:
For some reason, in the PR review I'm not getting the images loaded. If you want to visualize the doc, you can do so here: https://github.com/joperezr/aspire/blob/workloads/MSBuildSDKs/docs/specs/future-of-workload.md